store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -582,7 +584,7 @@ private String generateKey(StringBuilder keyBuilder, MessageExt messageExt) {
         if (tranType == MessageSysFlag.TRANSACTION_NOT_TYPE
                 || tranType == MessageSysFlag.TRANSACTION_COMMIT_TYPE) {
             // Delay Delivery
-            if (msg.getDelayTimeLevel() > 0) {
+            if (msg.getDelayTimeLevel() > 0 && !isDLQTopic(msg.getTopic())) {
```

Comment:
We could remove the delay_level property of DLQ msg from the source

Reply:
Seems a better way when the delayTimeLevel prop is useless after resend to DLQ topic.
I'll check it. Thanks for your review @yuz10 ~